nxcomp/Log.cpp

```diff
@@ -0,0 +1,81 @@
+
+#include <stdio.h>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+
+#include "Log.h"
+
+NXLog nx_log;
+
+
+NXLog::NXLog(char *filename)
+{
+	std::fstream *file = new std::fstream();
+	file->open(filename);
+	stream_ = file;
+	level_ = DEBUG;
+}
+
+bool NXLog::will_log() const
+{
+	std::map<std::string, NXLogLevel>::const_iterator item = per_file_levels_.find(current_file());
+
+	if ( item != per_file_levels_.end() )
+	{
+		return current_level_ >= item->second;
+	}
+	else
+	{
+		return current_level_ >= level_;
+	}
+}
+
+
+std::string NXLogStamp::to_string() const
+{
+	static const char* level_names[] = {
+		"DBG",
+		"INF",
+		"WRN",
+		"ERR",
+		"FAT"
+	};
+
+	std::ostringstream oss;
+	struct tm *timeinfo;
+	char buffer[32];
+	char buffer2[32];
+
+	oss << ((level_ >=0 && level_ < NXLOG_LEVEL_COUNT ) ? level_names[level_] : "???");
+
+
+	timeinfo = localtime(&timestamp_.tv_sec);
```

Comment:
Careful, `localtime` may not thread-safe... and there seems to be no cross-plattform alternative.

SUS has `localtime_r`, while Microsoft's `localtime` in VS 2005 and higher is supposed to be thread-safe.

Microsoft at least claims this:

>Both the 32-bit and 64-bit versions of gmtime, mktime, mkgmtime, and localtime all use a single tm structure per thread for the conversion.

Logging-code should really be thread-safe.